### PR TITLE
Add document for ClusterID

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
Signed-off-by: changzhen <changzhen5@huawei.com>

**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

Add document for [ClusterID](https://github.com/karmada-io/karmada/tree/master/docs/proposals/clusterid).

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
cc @Poor12 @yy158775 @RainbowMango

**Review by Arhell — COMMENTED** [review]
/assign @RainbowMango

File: docs/userguide/clustermanager/cluster-registration.md [hunk 1]
```diff
@@ -18,6 +18,10 @@ Each `karmada-agent` serves for a cluster and take responsibility for:
 - Maintains cluster status and reports to Karmada(updates the status of `Cluster` object)
 - Watch manifests from Karmada execution space(namespace, `karmada-es-<cluster name>`) and deploy to the cluster which serves for.
 
+## Cluster Identifier
+
+Regardless of the mode of cluster, in the progress of registering the cluster into Karmada, we will set the `.spec.id` field, which is different from the object uid(.metadata.uid), to confirm the cluster identity. You can refer to [Adoption of ClusterId](https://github.com/karmada-io/karmada/tree/master/docs/proposals/clusterid) for more details.
```
Comment (RainbowMango) [inline docs/userguide/clustermanager/cluster-registration.md:23]: Who is `we`?

Comment (XiShanYongYe-Chang) [reply docs/userguide/clustermanager/cluster-registration.md:23]: the registering progress?

**Comment by RainbowMango** [comment]
My suggestion:
```
Each cluster registered in Karmada will be represented as a `Cluster` object whose name(`.metadata.name`) is the 
registered name. The name will be widely used in the propagating process, such as specifying the location where a resource
should be propagated to in `PropagationPolicy`.

In addition, during the registration process, each cluster will be assigned a `unique identifier` which will be present at
the `.spec.id` of the `Cluster` object. For now, this `unique identifier` is used to distinguish each cluster technically 
to avoid accidentally registering the same cluster multiple times but using different register names.
The `unique identifier` is collected from the registering cluster's `kube-system` ID(`.metadata.uid`). 
```
Please @Tingtal take a look at this from a grammar point of view.

And please put this content at the end of this article, because people don't need to get involved before registering a cluster.

**Comment by XiShanYongYe-Chang** [comment]
> My suggestion:
> 
> ```
> Each cluster registered in Karmada will be represented as a `Cluster` object whose name(`.metadata.name`) is the 
> registered name. The name will be widely used in the propagating process, such as specifying the location where a resource
> should be propagated to in `PropagationPolicy`.
> 
> In addition, during the registration process, each cluster will be assigned a `unique identifier` which will be present at
> the `.spec.id` of the `Cluster` object. For now, this `unique identifier` is used to distinguish each cluster technically 
> to avoid accidentally registering the same cluster multiple times but using different register names.
> The `unique identifier` is collected from the registering cluster's `kube-system` ID(`.metadata.uid`). 
> ```
> 
> Please @Tingtal take a look at this from a grammar point of view.
> 
> And please put this content at the end of this article, because people don't need to get involved before registering a cluster.

Thanks~
/cc @Tingtal

File: i18n/zh/docusaurus-plugin-content-docs/current/userguide/clustermanager/cluster-registration.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ Each `karmada-agent` serves for a cluster and take responsibility for:
 - Maintains cluster status and reports to Karmada(updates the status of `Cluster` object)
 - Watch manifests from Karmada execution space(namespace, `karmada-es-<cluster name>`) and deploy to the cluster which serves for.
```
[the remark this answers is not in the document]
Comment (Tingtal) [reply i18n/zh/docusaurus-plugin-content-docs/current/userguide/clustermanager/cluster-registration.md:19]: This line "Each `karmada-agent` serves for a cluster and take responsibility for:" should be changed to:
**Each `karmada-agent` serves a cluster and:**

Seems I dont have the edit permission. Pls make the change.

**Review by Tingtal — CHANGES_REQUESTED** [review]
Language check.

**Comment by XiShanYongYe-Chang** [comment]
@Tingtal Thanks a lot~

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**